src/storage/settings/settingsSlice.ts

```diff
@@ -46,15 +47,22 @@ import SettingsStorage from '@/storage/settings/settingsPersistentStorage'
 import { RootState } from '@/storage/store'
 import { GeneralSettings } from '@/types/settings'
 
-const initialState: GeneralSettings = SettingsStorage.load('general') as GeneralSettings
+interface SettingsState extends GeneralSettings {
+  faucetCallPending: boolean
+}
+
+const initialState: SettingsState = {
+  ...(SettingsStorage.load('general') as GeneralSettings),
+  faucetCallPending: false
+}
 
 const settingsSlice = createSlice({
   name: 'settings',
   initialState,
   reducers: {},
   extraReducers(builder) {
     builder
-      .addCase(localStorageDataMigrated, () => SettingsStorage.load('general') as GeneralSettings)
+      .addCase(localStorageDataMigrated, () => SettingsStorage.load('general') as SettingsState)
```

Comment:
Hm, why this change? The type of the data loaded is `GeneralSettings`. `faucetCallPending` is not stored there.

Reply:
Because otherwise the compiler will complain, as `addCase` gets the expected type from `initialState`.
But this will be solved if I move `faucetCallPending ` to the `globalSlice` as you proposed below.